From 1f90ba995c8f22d4e69b347f2e3a49c2949024d8 Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Wed, 9 Mar 2016 10:45:48 +1000
Subject: [PATCH xserver] Xi: don't deliver emulated motion events for
 non-emulating touches

The touchpoint knows whether it should be emulating or not and we have a check
for that later. Check for this before we generate the event and try to deliver
it, lest we trigger a bug warning.

https://bugzilla.redhat.com/show_bug.cgi?id=1282252

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
---
 Xi/exevents.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Xi/exevents.c b/Xi/exevents.c
index 74e49ed..5a0b68d 100644
--- a/Xi/exevents.c
+++ b/Xi/exevents.c
@@ -1379,6 +1379,9 @@ DeliverTouchEmulatedEvent(DeviceIntPtr dev, TouchPointInfoPtr ti,
     if (!TouchResourceIsOwner(ti, listener->listener))
         return !Success;
 
+    if (!ti->emulate_pointer)
+        return !Success;
+
     nevents = TouchConvertToPointerEvent(ev, &motion, &button);
     BUG_RETURN_VAL(nevents == 0, BadValue);
 
-- 
2.5.0

